qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -174,30 +179,40 @@ def show_preview(self):
 
     def download_resource(self):
         resource = self.selected_resource()
-        file_path = download_resource_file(
-            resource.file, resource.resource_type, resource.uuid
-        )
+        file_extension = os.path.splitext(resource.file)[1]
+
+        file_path = QFileDialog.getSaveFileName(
+            self,
+            self.tr("Save Resource"),
+            resource.file,
+            self.tr(
+                "All Files (*);;Geopackage (*.gpkg);;QGIS Model (*.model3);; ZIP Files (*.zip)"
+            ),
+        )[0]
+
         if file_path:
-            text = f"Successfully download {resource.name} to {file_path}"
+            if not file_path.endswith(file_extension):
+                file_path = file_path + file_extension
+
+            download_resource_file(resource.file, file_path)
```

Comment:
You should check the return value of the `download_resource_file` to see whether the download is succeeded or not.